src/Microsoft.DotNet.Build.Tasks/GetPackageVersion.cs

```diff
@@ -34,7 +35,13 @@ public override bool Execute()
                 select el).FirstOrDefault();
 
             Debug.Assert(embeddedVerNode != null, "embeddedVerNode shouldn't be null");
-            PackageVersion = string.Format("{0}-{1}", embeddedVerNode.Value, BuildNumber);
+
+            // if this is a prerelease version then append the build number to the end
+            Regex preReleaseMatch = new Regex("-[0-9A-Za-z]+$");
```

Comment:
Sounds good I do think we will need to either keep the value in the nuspec or in the nuproj if we start using those to build the packages.